### ISSUE TITLE
Disconnecting when phone screen is shut off.

### ISSUE THREAD

**Issue by amoy1110** [shamblett/mqtt_client#44]
I have an app that is running in the background, but when you close the phone's screen by pressing the power button, it no longer sends the PINGREQ to the broker and disconnects.

How would I be able to keep the connection alive even when the phone's screen is turned off when my app is running in the background?

**Comment by shamblett** [comment]
The pingreq is sent on a simple timer expiry, if its not being sent then presumably the timer is not expiring. I don't know what happens when you close the phones screen, you could turn logging on to see if log output stops when you turn your screen off and starts again when you turn it on, if this is the case then your program isn't really running in the background, it may be alive but suspended somehow, either way the log should tell you whats happening.

**Comment by amoy1110** [comment]
I initially built an apk file and transferred it to my phone and installed it. When I did this, it would disconnect when I shut the screen off.
Then I tried it with my phone tethered to the computer and ran the app so I could see the log output, but it did not disconnect when I shut the screen off.
Does this mean that flutter does not allow the program to run code in the background?
I found this issue which does not seem to have a clear solution. https://github.com/flutter/flutter/issues/3671
